Sources/AsyncAlgorithms/AsyncZipLatestFromSequence.swift

```diff
@@ -0,0 +1,382 @@
+  /// that its latest know element is the one that has just been produced when the base sequence produces its next element.
+  ///
+  /// ```
+  /// let base = AsyncChannel<Int>()
+  /// let other = AsyncChannel<String>()
+  /// let sequence = base.zipLatest(from: other)
+  ///
+  /// Task {
+  ///   for element in await sequence {
+  ///    print(element)
+  ///   }
+  /// }
+  ///
+  /// await other.send("a")
+  /// await other.send("b")
+  ///
+  /// ... later in the application flow
+  ///
+  /// await base.send(1)
+  ///
+  /// // will print: (1, "b")
+  /// ```
+  ///
+  /// - Parameter other: the other ``AsyncSequence``
+  /// - Returns: an ``AsyncZipLatestFromSequence`` where elements are a tuple of an element from `self` and the
+  /// latest known element (if any) from the `other` sequence.
+  @inlinable
+  func zipLatest<Other: AsyncSequence>(
+    from other: Other
+  ) -> AsyncZipLatestFromSequence<Self, Other> {
+    AsyncZipLatestFromSequence(self, other)
+  }
+
+  /// Combines `self` with two other ``AsyncSequence`` into a single ``AsyncSequence`` where each
+  /// element from `self` is aggregated to the latest known elements from the `other` sequences (if any) as a tuple.
+  ///
+  /// Remark: as the `other` sequences are being iterated over in the context of their own ``Task``, there is no guarantee
+  /// that their latest know elements are the ones that have just been produced when the base sequence produces its next element.
+  ///
+  /// ```
+  /// let base = AsyncChannel<Int>()
+  /// let other1 = AsyncChannel<String>()
+  /// let other2 = AsyncChannel<String>()
+  /// let sequence = base.zipLatest(from: other1, other2)
+  ///
+  /// Task {
+  ///   for element in await sequence {
+  ///    print(element)
+  ///   }
+  /// }
+  ///
+  /// await other1.send("a")
+  /// await other1.send("b")
+  ///
+  /// await other2.send("c")
+  /// await other2.send("d")
+  ///
+  /// ... later in the application flow
+  ///
+  /// await base.send(1)
+  ///
+  /// // will print: (1, "b", "d")
+  /// ```
+  ///
+  /// - Parameters:
+  ///   - other1: the first other ``AsyncSequence``
+  ///   - other2: the second other ``AsyncSequence``
+  /// - Returns: an ``AsyncZipLatestFrom2Sequence`` where elements are a tuple of an element from `self` and the
+  /// latest known elements (if any) from the `other` sequences.
+  @inlinable
+  func zipLatest<Other1: AsyncSequence, Other2: AsyncSequence>(
+    from other1: Other1,
+    _ other2: Other2
+  ) -> AsyncZipLatestFrom2Sequence<Self, Other1, Other2> {
+    AsyncZipLatestFrom2Sequence(self, other1, other2)
+  }
+}
+
+/// ``AsyncZipLatestFromSequence`` is an ``AsyncSequence`` where elements are a tuple of an element from `base` and the
+/// latest known element (if any) from the `other` sequence.
+public struct AsyncZipLatestFromSequence<Base: AsyncSequence, Other: AsyncSequence>: AsyncSequence where Other: Sendable {
```

Comment:
these types probably need to be audited for being sendable. conditionally or not; my guess is that they will have similar constraints to zip/combineLatest.